authentication/authentication.go

```diff
@@ -1,13 +1,113 @@
+	Locale string `json:"locale,omitempty"`
+	// Middle name(s) of the user.
+	MiddleName string `json:"middle_name,omitempty"`
+	// Full name of the user in displayable form including all name parts, possibly including titles and
+	// suffixes, ordered according to the user's locale and preferences.
+	Name string `json:"name,omitempty"`
+	// Casual name of the user that may or may not be the same as FirstName.
+	Nickname string `json:"nickname,omitempty"`
+	// The user's preferred telephone number.
+	PhoneNumber string `json:"phone_number,omitempty"`
+	// Whether the user's phone number has been verified or not.
+	PhoneNumberVerified bool `json:"phone_number_verified,omitempty"`
+	// URL of the user's profile picture.
+	Picture string `json:"picture,omitempty"`
+	// Shorthand name by which the user wishes to be referred by, such as janedoe or j.doe.
+	PreferredUsername string `json:"preferred_username,omitempty"`
+	// URL of the user's profile page.
+	Profile string `json:"profile,omitempty"`
+	// Subject-Identifier for the user at the issuer. A unique value to identify the user.
+	Sub string `json:"sub,omitempty"`
+	// Time and date the user's information was last updated.
+	UpdatedAt *time.Time `json:"updated_at"`
+	// URL of the user's Web page or blog.
+	Website string `json:"website,omitempty"`
+	// User's time zone as a "tz database name".
+	ZoneInformation string `json:"zoneinfo,omitempty"`
+}
+
+// UserAddress defines a user's address.
+type UserAddress struct {
+	// Country component of the address.
+	Country string `json:"country,omitempty"`
+	// Full mailing address, formatted for display or use on a mailing label.
+	Formatted string `json:"formatted,omitempty"`
+	// City or locality component of the address.
+	Locality string `json:"locality,omitempty"`
+	// Zip or postal code component of the address.
+	PostalCode string `json:"postal_code,omitempty"`
+	// State, province, prefecture, or region component of the address.
+	Region string `json:"region,omitempty"`
+	// Full street address component, which may include house number, street name, Post Office Box,
+	// and multi-line extended street address information.
+	StreetAddress string `json:"street_address,omitempty"`
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+//
+// It is required to handle the mapping of unknown claims from the main response body into the AdditionalClaims
+// field on the struct.
+func (u *UserInfoResponse) UnmarshalJSON(b []byte) error {
+	type userInfoWrapper UserInfoResponse
+	ui := userInfoWrapper{}
+
+	err := json.Unmarshal(b, &ui)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(b, &(ui.AdditionalClaims))
+	if err != nil {
+		return err
+	}
+
+	typ := reflect.TypeOf(ui)
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		jsonTag := strings.Split(field.Tag.Get("json"), ",")[0]
+		if jsonTag != "" && jsonTag != "-" {
+			delete(ui.AdditionalClaims, jsonTag)
+		}
+	}
```

Comment:
Could this be done using the structs package?

Reply:
I think if we added extra tags with the `structs` tag that duplicated the json tags it potentially could